tweets/views.py

```diff
@@ -39,3 +43,47 @@ def test_func(self):
             return current_user.pk == tweet_user.pk
         else:
             raise Http404
+
+
+@login_required
+@require_POST
+def like_view(request, pk):
+    tweet = get_object_or_404(Tweet, pk=pk)
+    user = request.user
+    liked = False
+    like = Like.objects.filter(tweet=tweet, user=user)
+    if like.exists():
+        pass
+    else:
+        like.create(tweet=tweet, user=user)
+        liked = True
+
+    context = {
+        "tweet_id": tweet.id,
+        "liked": liked,
+        "count": tweet.like_set.count(),
+    }
+
+    return JsonResponse(context)
+
+
+@login_required
+@require_POST
+def unlike_view(request, pk):
+    tweet = get_object_or_404(Tweet, pk=pk)
+    user = request.user
+    liked = False
+    like = Like.objects.filter(tweet=tweet, user=user)
+    if like.exists():
+        like.delete()
+        liked = False
+    else:
+        pass

```

Comment:
python3.8以上であれば、セイウチ演算子というものが使えて、以下のように短縮して書けます。
```suggestion
    if like := Like.objects.filter(tweet=tweet, user=request.user):
        like.delete()
    liked = False
```

Reply:
修正しました。